app/uk/gov/hmrc/helptosavefrontend/controllers/NewApplicantUpdateEmailAddressController.scala

```diff
@@ -135,16 +136,27 @@ class NewApplicantUpdateEmailAddressController @Inject() (val sessionCacheConnec
             logger.warn(e)
             internalServerError()
         }, { maybeNSIUserInfo ⇒
-          maybeNSIUserInfo.fold(
-            _ ⇒ SeeOther(routes.EligibilityCheckController.getIsNotEligible().url),
-            _ ⇒ SeeOther(routes.NewApplicantUpdateEmailAddressController.getEmailVerified().url)
-          )
+          maybeNSIUserInfo.fold({
+            createAccountVoid ⇒
+              createAccountVoid.result.fold(
+                _ ⇒ SeeOther(routes.NewApplicantUpdateEmailAddressController.getLinkExpiredPage().url),
```

Comment:
in this case , the link expired response is confusing, -  since the user has created an account already , can't we say something like - `you have already verified your email and account has been created already , please go to your account page... blah blah...`
`getLinkExpiredPage` actually makes sense if users have delayed verifying their email and so the link has expired ...

Reply:
I think we need to ask Jean about this??